### PR TITLE
Add basic UDP server

### PR DESCRIPTION
**Pull request by LukasJoswiak**

### CODE REVIEW

File: include/UDPClient.hpp [hunk 1]
```diff
@@ -12,14 +12,19 @@ using boost::asio::ip::udp;
...
 
  private:
+  void StartRead();
+  void HandleRead(const boost::system::error_code &ec,
+      std::size_t bytes_transferred);
+
   udp::socket socket_;
-  udp::endpoint endpoint_;
+  udp::endpoint remote_endpoint_;
+  std::array<char, 1024> recv_buffer_;
```
Comment (TheJMaster) [inline include/UDPClient.hpp]: nit: no magic numbers

File: include/UDPServer.hpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  void HandleRead(const boost::system::error_code &ec,
+      std::size_t bytes_transferred);
+
+  void HandleSend(const boost::system::error_code &ec,
+      std::size_t bytes_transferred);
+
+ private:
+  udp::socket socket_;
+  udp::endpoint remote_endpoint_;
+  std::array<char, 1024> recv_buffer_;
```
Comment (TheJMaster) [inline include/UDPServer.hpp]: nit: no magic numbers

File: src/udp/server/UDPServer.cpp [hunk 3]
```diff
@@ -1,8 +1,57 @@
...
+
+UDPServer::UDPServer(boost::asio::io_context &io_context, int port)
+    : socket_(io_context, udp::endpoint(udp::v4(), port)) {
+  StartRead();
+}
+
+void UDPServer::StartRead() {
+  socket_.async_receive_from(boost::asio::buffer(recv_buffer_),
+      remote_endpoint_, boost::bind(&UDPServer::HandleRead, this,
+        boost::asio::placeholders::error,
```
Comment (TheJMaster) [inline src/udp/server/UDPServer.cpp:17]: double indent?

File: src/udp/server/UDPServer.cpp [hunk 4]
```diff
@@ -1,8 +1,57 @@
...
+  if (!ec) {
+    auto message = std::string(&recv_buffer_[0],
+        &recv_buffer_[0] + bytes_transferred);
+    std::cout << "Received message: " << message << std::endl;
+
+    std::string response = "Hello, Client!";
+
+    socket_.async_send_to(boost::asio::buffer(response), remote_endpoint_,
+        boost::bind(&UDPServer::HandleSend, this,
+          boost::asio::placeholders::error,
```
Comment (TheJMaster) [inline src/udp/server/UDPServer.cpp:32]: double indent?

**Review by TheJMaster — APPROVED** [review]

File: src/udp/server/UDPServer.cpp (hunk 4 above)
Comment (LukasJoswiak) [reply src/udp/server/UDPServer.cpp:32]: The second indent is due to calling another function, `bind`, inside the first indent.

File: src/udp/server/UDPServer.cpp (hunk 3 above)
Comment (LukasJoswiak) [reply src/udp/server/UDPServer.cpp:17]: See below.

**Pull request merged**